spacelift/resource_named_webhook_secret_header.go

```diff
@@ -0,0 +1,152 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/shurcooL/graphql"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/validations"
+)
+
+func resourceNamedWebhookSecretHeader() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_named_webhook_secret_header` represents secret key value combination used as a custom header" +
+			"when delivering webhook requests. It depends on `spacelift_named_webhook` resource which should exist.",
+
+		CreateContext: resourceNamedWebhookSecretHeaderCreate,
+		ReadContext:   resourceNamedWebhookSecretHeaderRead,
+		DeleteContext: resourceNamedWebhookSecretHeaderDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"webhook_id": {
+				Type:        schema.TypeString,
+				Description: "ID of the stack on which the environment variable is defined",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"key": {
+				Type:             schema.TypeString,
+				Description:      "key for the header",
+				Required:         true,
+				ValidateDiagFunc: validations.DisallowEmptyString,
+				ForceNew:         true,
+			},
+			"value": {
+				Type:             schema.TypeString,
+				Description:      "value for the header",
+				DiffSuppressFunc: suppressValueChange,
+				Required:         true,
+				Sensitive:        true,
+				ForceNew:         true,
+			},
+		},
+	}
+}
+
+func resourceNamedWebhookSecretHeaderCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var mutation struct {
+		WebhooksIntegration struct {
+			ID string `graphql:"id"`
+		} `graphql:"namedWebhooksIntegrationSetHeaders(id: $id, input: $input)"`
+	}
+
+	webhookID, _ := d.GetOk("webhook_id")
+	variables := map[string]interface{}{
+		"id": toID(webhookID),
+		"input": structs.NamedWebhookHeaderInput{
+			Entries: []structs.KeyValuePair{
+				{
+					Key:   d.Get("key").(string),
+					Value: d.Get("value").(string),
+				},
+			},
+		},
+	}
+
+	var ret diag.Diagnostics
+	if err := meta.(*internal.Client).Mutate(ctx, "NamedWebhookSecretHeaderSet", &mutation, variables); err != nil {
+		ret = diag.Errorf("could not set secret header: %v", internal.FromSpaceliftError(err))
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", d.Get("webhook_id"), d.Get("key")))
+
+	return append(ret, resourceNamedWebhookSecretHeaderRead(ctx, d, meta)...)
+}
+
+func resourceNamedWebhookSecretHeaderRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
If the changes happen behind the scenes (eg. manually changed in the GU), these will not be detected by Terraform. Within the current API I don't see a way of addressing it, and it's probably not a big deal in the 1st place, but in the future it would be great to see API design that is more Terraform-friendly.

Reply:
Good point, will take that in to account in the future. It ended up being a strange API design all together tbh. In the end I'm not super happy about it either. :/

Reply:
It's not a high traffic resource so it's probably fine.